cleanlab/internal/neighbor/metric.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+from scipy.spatial.distance import euclidean
+
+from cleanlab.internal.neighbor.types import Metric
+
+HIGH_DIMENSION_CUTOFF: int = 3
+"""
+The cutoff value for the number of columns in the feature array.
+
+If the number of columns (M) in the feature array is greater than this cutoff value,
+the "cosine" metric is used. This is because the cosine metric is more suitable for
+high-dimensional data.
+"""
+ROW_COUNT_CUTOFF: int = 100
+"""
+The cutoff value for the number of rows in the feature array. 
+
+If the number of rows (N) in the feature array is greater than this cutoff value,
+the "euclidean" metric is used (implemented in scikit-learn for efficiency reasons).
+Otherwise, the implementation of the euclidean metric from scipy is used, which is much slower but
+more accurate in general.
+"""
```

Comment:
```suggestion
HIGH_DIMENSION_CUTOFF: int = 3
"""
If the number of columns (M) in the `features` array is greater than this cutoff value,
then by default K-nearest-neighbor operations will use the "cosine" metric. The cosine metric is more suitable for high-dimensional data.
"""
ROW_COUNT_CUTOFF: int = 100
"""
The cutoff value for the number of rows in the feature array. 

If the number of rows (N) in the feature array is greater than this cutoff value,
the "euclidean" metric is used (implemented in scikit-learn for efficiency reasons).
Otherwise, the implementation of the euclidean metric from scipy is used, which is much slower but
more accurate in general.
"""
```